### PR TITLE
Feature/issue 13 result manager should store results scoped per poll

### PR DESCRIPTION
**Pull request by dmaclay**

### CODE REVIEW

File: tests/test_multipoll_example.py [hunk 1]
```diff
@@ -778,3 +778,82 @@ def test_register_1_archive_and_repeat(self):
...
+                },
+                {
+                'checks': {'equal': {'TEST': '2'}},
+                'copy': "Thats negative.",
+                'valid_responses': [],
+                'label': '',
+                }]
+
+    @inlineCallbacks
+    def setUp(self):
```
Comment (hodgestar) [inline tests/test_multipoll_example.py:813]: setUp and run_inputs appear to have been duplicated a number of times. Could you create a ticket for factoring these methods out into a common base class?

File: vxpolls/manager.py [hunk 2]
```diff
@@ -190,33 +190,31 @@ def get_next_question(self, participant, last_index=None):
 
     def is_suitable_question(self, participant, question):
 
-        state = self.results_manager.get_user(self.poll_id,
-                        participant.user_id)
-        extended_state = dict(state, **participant.labels)
+        state = participant.labels
 
         def equals(key, value):
-            return unicode(extended_state[key]) == unicode(value)
```
Comment (hodgestar) [inline vxpolls/manager.py:198]: The changes from [key] to .get(key) are a behavioural change. I think .get(key) returning None is better than getting a KeyError though so I'm +1 on the change.

File: vxpolls/manager.py [hunk 3]
```diff
@@ -190,33 +190,31 @@ def get_next_question(self, participant, last_index=None):
 
     def is_suitable_question(self, participant, question):
 
-        state = self.results_manager.get_user(self.poll_id,
-                        participant.user_id)
-        extended_state = dict(state, **participant.labels)
+        state = participant.labels
```
Comment (hodgestar) [inline vxpolls/manager.py:193]: I guess this is the crucial part of the change and I'm not very familiar with vxpolls, so I have some questions / observations:
- Is it possible for `state` to have keys not in `participant.labels`?
- If not, does this PR fix a bug? (in this case `extended_state` and `participant.labels` are identical)
- If yes, is it safe to exclude these keys?

Comment (dmaclay) [inline vxpolls/manager.py:193]: - Definitely possibly for state to have keys not in participant.labels (1/2 the root of the problem)
- N/A
- Yes, hence the switch from dict[key] to dict.get(key)

**Comment by jerith** [comment]
:+1: as long as you're happy.

**Comment by hodgestar** [comment]
:+1: on landing this.

**Pull request merged**